src/host.c

```diff
@@ -626,6 +626,28 @@ cache_store (const char *host, struct address_list *al)
     }
 }
 
+void 
+cache_store (const char *host, const char *ip)
+{
+  if (!is_valid_ipv4_address(ip, ip + strlen(ip))) {
+    logputs(LOG_ALWAYS, "Not a valid IP address: %s", ip);
+    abort();
+  }
+
+  struct address_list *al = xnew0 (struct address_list);
+
+  al->addresses = xnew_array (ip_address, 1);
+  al->count     = 1;
+  al->refcount  = 1;
+
+  ip_address *addr = &al->addresses[0];
+  addr->family = AF_INET;
+  memcpy (IP_INADDR_DATA (addr), ip, 4);
+
+  cache_store(host, al);
```

Comment:
recursive call? Why?

Reply:
right, there's no parametric polymorphism in C.

Reply:
Thanks!